pkg/activator/handler/handler.go

```diff
@@ -54,5 +54,28 @@ func (a *ActivationHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	proxy := httputil.NewSingleHostReverseProxy(target)
 	proxy.Transport = a.Transport
 
+	setupPruningHeaders(proxy)
+
 	proxy.ServeHTTP(w, r)
 }
+
+func setupPruningHeaders(p *httputil.ReverseProxy) {
+	headersToRemove := []string{
+		activator.ConfigurationHeader,
+		activator.RevisionHeaderName,
+		activator.RevisionHeaderNamespace,
+	}
+
+	orig := p.Director
+	p.Director = func(r *http.Request) {
+		if orig != nil {
+			orig(r)
+		}
+
+		for _, h := range headersToRemove {
+			if r.Header.Get(h) != "" {
+				r.Header.Del(h)
```

Comment:
I think you can omit the surrounding `if` since `Del` will delegate to `delete` which is a noop if nothing is there to delete.